src/world/Server/Packets/Handlers/MovementHandler.cpp

```diff
@@ -177,31 +177,32 @@ static MovementFlagName MoveFlagsToNames[] =
 static const uint32 nmovementflags = sizeof(MoveFlagsToNames) / sizeof(MovementFlagName);
 void WorldSession::HandleMovementOpcodes(WorldPacket& recv_data)
 {
-    uint16 opcode = recv_data.GetOpcode();
+    uint32 opcode = recv_data.GetOpcode();
     Player* mover = _player;
 
     if (MoverGuid != mover->GetGUID())
         return;
 
-    if (_player->GetCharmedByGUID() || !_player->IsInWorld() || _player->GetPlayerStatus() == TRANSFER_PENDING || _player->GetTaxiState())
+    if (mover->GetCharmedByGUID() || !mover->IsInWorld() || mover->GetPlayerStatus() == TRANSFER_PENDING || mover->GetTaxiState())
     {
         return;
     }
 
     /************************************************************************/
     /* Clear standing state to stand.				                        */
     /************************************************************************/
-    if (recv_data.GetOpcode() == CMSG_MOVE_START_FORWARD)
-        _player->SetStandState(STANDSTATE_STAND);
+    if (opcode == CMSG_MOVE_START_FORWARD)
```

Comment:
What happen if a player turn left or move backward?
Is this really the correct place to set standstate?